core/sdk-core/src/main/java/software/amazon/awssdk/core/protocol/MarshallingType.java

```diff
@@ -69,20 +69,28 @@ public interface MarshallingType<T> {
 
     Class<? super T> getTargetClass();
 
+    default MarshallingKnownType getKnownType() {
```

Comment:
I might be missing something , while doing a get we do below
```
        public Object get(MarshallingType<?> marshallingType) {
            MarshallingKnownType knownType = marshallingType.getKnownType();
            if (knownType != null) {
                return l1registry.get(knownType);
            }
            return l2registry.get(marshallingType);
        }
```

since predefined  (line 38 to 68) MarshallingType are also predefined in MarshallingKnownType , what cases we will end up in predefined type not present in l1registry but present in l2registry ?
I was thinking if we can (line 38 to 68) is redendant